### PR TITLE
Popup menu #917

### PR DESCRIPTION
**Pull request by ksheremet**

### CODE REVIEW

**Comment by ksheremet** [comment]
ezgif com-video-to-gif (3)

File: flutter/lib/views/decks_list/deck_menu.dart [hunk 1]
```diff
@@ -0,0 +1,285 @@
...
+    }
+  }
+}
+
+enum _DeckMenuItemType { add, edit, setting, share }
+
+Map<_DeckMenuItemType, String> _buildMenu(BuildContext context) {
+  // We want this Map to be ordered.
+  // ignore: prefer_collection_literals
+  final deckMenu = LinkedHashMap<_DeckMenuItemType, String>()
```
Comment (dotdoom) [inline flutter/lib/views/decks_list/deck_menu.dart]: Map literals are ordered, can use `{}` instead.

File: flutter/lib/views/decks_list/deck_menu.dart [hunk 2]
```diff
@@ -0,0 +1,285 @@
...
+    super.initState();
+    _controller = widget.controller;
+
+    final anim = CurvedAnimation(parent: _controller, curve: Curves.linear);
+    _moveAnimation = Tween<Alignment>(
+            begin: Alignment.centerRight, end: Alignment.bottomRight)
+        .animate(anim);
+    // Tween animation doesn't support int literals
+    // ignore: prefer_int_literals
+    _opacityAnimation = Tween(begin: 0.0, end: 1.0).animate(anim);
```
Comment (dotdoom) [inline flutter/lib/views/decks_list/deck_menu.dart]: Perhaps `Tween<double>(begin: 0, end: 0)` would show the intent better, and won't need a linter override?

File: flutter/lib/views/decks_list/deck_menu.dart [hunk 3]
```diff
@@ -0,0 +1,285 @@
...
+  void _onMenuClosed() {
+    setState(() {
+      _menuIcon = _getMenuIcon(Theme.of(context).platform);
+    });
+  }
+
+  @override
+  Widget build(BuildContext context) => IconButton(
+        icon: Icon(
+          _menuIcon ?? _getMenuIcon(Theme.of(context).platform),
```
Comment (dotdoom) [inline flutter/lib/views/decks_list/deck_menu.dart]: Is caching necessary, and does it work fine with iOS / Android mode switch?

File: flutter/lib/views/decks_list/deck_menu.dart [hunk 4]
```diff
@@ -0,0 +1,285 @@
...
+  if (!CurrentUserWidget.of(context).user.isAnonymous) {
+    deckMenu[_DeckMenuItemType.share] =
+        AppLocalizations.of(context).shareDeckMenu;
+  }
+  return deckMenu;
+}
+
+class _MenuRoute<_DeckMenuItemType> extends PopupRoute<_DeckMenuItemType> {
+  // We need parent to count position of menu.
+  BuildContext parent;
```
Comment (dotdoom) [inline flutter/lib/views/decks_list/deck_menu.dart]: final?

File: flutter/lib/views/decks_list/deck_menu.dart [hunk 5]
```diff
@@ -0,0 +1,285 @@
...
+
+  AnimationController controller;
```
Comment (dotdoom) [inline flutter/lib/views/decks_list/deck_menu.dart]: final?

**Pull request merged**